backend/src/main/java/fk/prof/backend/model/assignment/impl/SimultaneousWorkAssignmentCounterImpl.java

```diff
@@ -0,0 +1,36 @@
+package fk.prof.backend.model.assignment.impl;
+
+import fk.prof.backend.model.assignment.SimultaneousWorkAssignmentCounter;
+import fk.prof.backend.proto.BackendDTO;
+import recording.Recorder;
+
+/**
+ * No operation of this class is thread-safe
+ * Right now, only invoked from backend daemon thread, so we are good
+ */
+public class SimultaneousWorkAssignmentCounterImpl implements SimultaneousWorkAssignmentCounter {
+  private final int maxAllowedConcurrentWorkAssignmentSlots;
+  private int currentlyAcquiredWorkAssignmentSlots = 0;
+
+  public SimultaneousWorkAssignmentCounterImpl(int maxAllowedConcurrentWorkAssignmentSlots) {
+    this.maxAllowedConcurrentWorkAssignmentSlots = maxAllowedConcurrentWorkAssignmentSlots;
+  }
+
+  /**
+   * TODO: Should be refactored to take process group and work prototype as input and accordingly determine slots
+   * @return
+   */
+  public int acquireSlots(Recorder.ProcessGroup processGroup, BackendDTO.WorkProfile workProfile) {
```

Comment:
Ah so this is a dummy impl. I have made it clear in the comments above this method. Ideally, the method should determine what is the min ||-ism required for a given [processgroup, workprofile]. Or do you think we should move out that functional logic to aggregation window planner?